webapp/tests/karma/js/enketo/widgets/dynamic-url.spec.ts

```diff
@@ -47,7 +47,9 @@ describe('Enketo: Dynamic URL Widget', () => {
     const dynamic = 'worldHello';
     $('.url span').text(dynamic);
 
-    expect($(DynamicUrlWidget.selector).attr('href')).to.equal(`http://google.com?q=${dynamic}`);
+    setTimeout(() => {
+      expect($(DynamicUrlWidget.selector).attr('href')).to.equal(`http://google.com?q=${dynamic}`);
+    }, 0);
```

Comment:
My take on this is you've made the expectation async, so the test thread will complete before the `expect` executes. This can lead to nasty issues where the test is marked as passed and then an error message is output in the log but doesn't cause CI to fail. While I don't see this happening in your CI build if this expectation was ever broken in the future then it probably wouldn't be picked up. You can test this out locally by changing the expected URL and making sure the test fails.

The fix for this is probably to add the `done` parameter to the function definition and calling `done()` inside the `setTimeout` callback.

```suggestion
    setTimeout(() => {
      expect($(DynamicUrlWidget.selector).attr('href')).to.equal(`http://google.com?q=${dynamic}`);
      done();
    }, 0);
```